src/app/features/Chanels/redux/actions.js

```diff
@@ -33,9 +44,45 @@ export function fetchPosts(id) {
       requestPosts(id).then((result) => {
         const postItem = result.find((item) => item.channelId === id);
         if (postItem) {
-          dispatch(getPosts(postItem));
+          dispatch(getPostsAction(postItem));
```

Comment:
please, return one post from **requestPosts** function. As soon as you provide channel id to this function, you don't have to find the posts in action creator.